### PR TITLE
Make bin/ in makefile, not configure.

### PR DESCRIPTION
**Pull request by SkySkimmer**
See https://github.com/coq/coq/pull/7984#issuecomment-402315772 for context.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -156,7 +156,10 @@ endif
 
 MAKE_OPTS := --warn-undefined-variable --no-builtin-rules
 
-submake: alienclean
+bin:
+	mkdir bin
+
+submake: alienclean | bin
```
Comment (ejgallego) [inline Makefile]: Umm, I guess the rules generating the binaries should depend on the `bin` target? Anyways I am not an expert in `Makefile`.

Also, shouldn't this use `mkdir -p bin`?

Comment (gares) [reply Makefile]:
```
$ mkdir a
$ mkdir a # error
```
I guess `-p` is needed but may not be spot by CI

Comment (SkySkimmer) [reply Makefile]: I tried making rules generating things in bin depend on it but could not figure out how to make it work in a generic way (`bin/%: bin` didn't have an effect) and doing it per-rule seems like a pain.

About -p, the rule shouldn't get run if the directory already exists.

Comment (gares) [reply Makefile]: ah right. then I'm OK with it.

**Review by gares — APPROVED** [review]

**Pull request merged**